Assets/MixedRealityToolkit.SDK/Features/Utilities/Solvers/SolverHandler.cs

```diff
@@ -10,45 +10,72 @@ namespace Microsoft.MixedReality.Toolkit.Utilities.Solvers
         [Tooltip("Tracked object to calculate position and orientation from. If you want to manually override and use a scene object, use the TransformTarget field.")]
-        private TrackedObjectType trackedObjectToReference = TrackedObjectType.Head;
+        private TrackedObjectType trackedTargetType = TrackedObjectType.Head;
 
         /// <summary>
         /// Tracked object to calculate position and orientation from. If you want to manually override and use a scene object, use the TransformTarget field.
         /// </summary>
-        public TrackedObjectType TrackedObjectToReference
+        public TrackedObjectType TrackedTargetType
         {
-            get { return trackedObjectToReference; }
+            get { return trackedTargetType; }
             set
             {
-                if (trackedObjectToReference != value)
+                if (trackedTargetType != value)
                 {
-                    trackedObjectToReference = value;
+                    trackedTargetType = value;
                     RefreshTrackedObject();
                 }
             }
         }
 
         [SerializeField]
-        [Tooltip("When using the hand to calculate position and orientation, use this specific joint.")]
+        [Tooltip("If tracking hands or motion controllers, determines which hand(s) are valid attachments")]
+        private Handedness trackedHandness = Handedness.Both;
+
+        /// <summary>
+        /// If tracking hands or motion controllers, determines which hand(s) are valid attachments
+        /// </summary>
+        public Handedness TrackedHandness
+        {
+            get { return trackedHandness; }
+            set
+            {
+                if (trackedHandness != value)
+                {
+                    trackedHandness = value;
+                    RefreshTrackedObject();
+                }
+            }
+        }
+
+        [SerializeField]
+        [Tooltip("When TrackedTargetType is set to hands, use this specific joint to calculate position and orientation")]
         private TrackedHandJoint trackedHandJoint = TrackedHandJoint.Palm;
 
         /// <summary>
-        /// When using the hand to calculate position and orientation, use this specific joint.
+        /// When TrackedTargetType is set to hands, use this specific joint to calculate position and orientation
         /// </summary>
         public TrackedHandJoint TrackedHandJoint
         {
             get { return trackedHandJoint; }
             set
             {
-                trackedHandJoint = value;
-                RefreshTrackedObject();
+                if (trackedHandJoint != value)
+                {
+                    trackedHandJoint = value;
+                    RefreshTrackedObject();
+                }
             }
         }
 
+        [SerializeField]
+        [Tooltip("Manual override for when TrackedTargetType is set to CustomOverride")]
+        private Transform transformOverride;
```

Comment:
>private Transform transformOverride; [](start = 8, length = 36)

Out of curiousity, why was the runtime setter for this property done as a function (i.e. SetTransformOverride) instead of as a property like the other ones above?

Reply:
Asking mostly out of consistency, not because one way is better than the other

---
In reply to: [308995399](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/5416#discussion_r308995399) [](ancestors = 308995399)

Reply:
Because there is no getter so I made a set function. But it does feel better to make it a public property setter instead